test/known_problems/should_pass/pattern_ann_type.erl

```diff
@@ -0,0 +1,13 @@
+-module(pattern_ann_type).
```

Comment:
I suggest you put this file addition in a separate patch.

Reply:
ok

(It is a simplification of the below error that comes out during self-gradualization, I thought that that error is related to the changes in this PR, but it turns out it is not
```
The pattern {var,_,'_'} on line {3345,14} doesn't have the type Pat :: gradualizer_type:af_pattern()
```
)